File: lab03_stack/lab03_example01/src/lab03_example01_pub.cpp
#include <ros/ros.h>
#include <example_msg_srv/test_msg.h>

double fRand(double fMin, double fMax)
{
    double f = (double)rand() / RAND_MAX;
    return fMin + f * (fMax - fMin);
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "rotation_publisher");

    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<example_msg_srv::test_msg>("/publish_rotation", 100);

    ros::Rate loop_rate(10);

    example_msg_srv::test_msg my_msg;

    while (nh.ok())
    {
        srand(time(0));
        my_msg.rotx.data = fRand(-2.0, 2.0);
        srand(time(0));
        my_msg.roty.data = fRand(-1.0, 1.0);
        srand(time(0));
        my_msg.rotz.data = fRand(-1.5, 1.5);

        pub.publish(my_msg);

        ros::spinOnce();

        loop_rate.sleep();
    }

}

File: lab03/lab03_example01/scripts/lab03_example01_pub.py
#!/usr/bin/env python

import rospy
import random
from example_msg_srv.msg import test_msg

def rot_publisher():

    rospy.init_node('rotation_publisher', anonymous=True)
    pub = rospy.Publisher('publish_rotation', test_msg, queue_size=100)
    rate = rospy.Rate(10)

    rot_msg = test_msg()

    while not rospy.is_shutdown():

        rot_msg.rotx.data = random.uniform(-2.0, 2.0)
        rot_msg.roty.data = random.uniform(-1.7, 1.7)
        rot_msg.rotz.data = random.uniform(-1.0, 1.8)

        pub.publish(rot_msg)
        rate.sleep()

if __name__ == '__main__':
    try:
        rot_publisher()
    except rospy.ROSInterruptException:
        pass